bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/EclipseIniUtil.java

```diff
@@ -0,0 +1,218 @@
+/*******************************************************************************
+ * Copyright 2021-2022 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.core.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections4.map.HashedMap;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.URIUtil;
+
+/**
+ * Utility class for managing and editing different parameter in eclipse.ini and config.ini file for eclipse
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class EclipseIniUtil
+{
+	private final String ECLIPSE_HOME_LOCATION;
+	private final String ECLIPSE_INI_FILE = "eclipse.ini";
+	public static final String ECLIPSE_INI_VMARGS = "-vmargs";
+
+	private List<String> eclipseIniFileContents;
+	private List<String> eclipseIniArgs;
+	private Map<String, String> eclipseIniSwitchMap;
+	private URI eclipseIniUri;
+	private Map<String, String> eclipseVmArgMap;
+
+	public EclipseIniUtil() throws Exception
+	{
+		ECLIPSE_HOME_LOCATION = System.getProperty("eclipse.home.location");
+		eclipseIniUri = URIUtil.fromString(ECLIPSE_HOME_LOCATION.concat(ECLIPSE_INI_FILE));
+		loadEclipseIniFileContents();
+		loadEclipseIniSwitchMap();
+		loadEclipseVmArgMap();
+	}
+
+	/**
+	 * Sets the eclipse switch in eclipse.ini file for eclipse, these are not vmargs. Changes will not take effect until
+	 * restart
+	 * 
+	 * @param eclipseSwitch The switch to set with the respective hyphen e.g: -nl or --launcher.defaultAction
+	 * @param switchValue   The value of the switch to set
+	 * @throws IOException
+	 */
+	public void setEclipseSwitchValue(String eclipseSwitch, String switchValue) throws IOException
+	{
+		eclipseIniSwitchMap.put(eclipseSwitch, switchValue);
+
+		updateEclipseIniFile();
+	}
+
+	/**
+	 * Removes the eclipse switch from eclipse.ini
+	 * 
+	 * @param eclipseSwitch The switch to remove with the respective hyphen e.g: -nl or --launcher.defaultAction
+	 * @throws IOException
+	 */
+	public void removeEclipseSwitch(String eclipseSwitch) throws IOException
+	{
+		eclipseIniSwitchMap.remove(eclipseSwitch);
+
+		updateEclipseIniFile();
+	}
+
+	/**
+	 * Gets the current value for a switch
+	 * 
+	 * @param eclipseSwitch The switch to get with the respective hyphen e.g: -nl or --launcher.defaultAction
+	 * @return
+	 */
+	public String getEclipseIniSwitchValue(String eclipseSwitch)
+	{
+		return eclipseIniSwitchMap.get(eclipseSwitch);
+	}
```

Comment:
Are we using this anywhere?

Reply:
Can be useful in the future when we may have to pass some other switches to the eclipse or read their value